pkg/storage/batch.go

```diff
@@ -139,6 +139,36 @@ func (r *BatchReader) Value() []byte {
 	}
 }
 
+// Verify verifies the checksums of the current record in the batch entry
+// will return error on mismatch.
+func (r *BatchReader) Verify() error {
+	key, err := r.EngineKey()
+	if err != nil {
+		return err
+	}
+	// TODO: verify values with other encoding types.
+	if !key.IsMVCCKey() {
+		return nil
+	}
+	mvccKey, err := key.ToMVCCKey()
+	if err != nil {
+		return err
+	}
+	if mvccKey.IsValue() {
```

Comment:
@erikgrinaker

What's with the `!mvccKey.IsValue()` case? As I understood, it means the timestamp is empty, and it's some kind of "meta" key/value. Do we not compute a checksum for these key/values?

What are examples of "meta" values? Intents?

Reply:
Intents and inline values. These contain an encoded `enginepb.MVCCMetadata`, with `Value.RawBytes` in `MVCCMetadata.Value`. These should be decoded to a `roachpb.Value` and verified too.

Reply:
Intents, i.e. lock-table keys, also use `MVCCMetadata` with a checksum in `RawBytes`.

I'm not sure if we ever use `PutUnversioned()` with anything that isn't an `MVCCMetadata` in the replicated keyspace. Could possibly be something in the range-local keyspace. I guess time-series data also won't be a `roachpb.Value`, but I'm not sure. We'll find out if something fails once we attempt to decode inline values.